src/itx.rs

```diff
@@ -137,3 +139,277 @@ pub unsafe extern "C" fn inv_txfm_add_rust<BD: BitDepth>(
         dst = dst.offset(BD::pxstride(stride) as isize);
     }
 }
+
+pub type itxfm_fn = Option<
+    unsafe extern "C" fn(*mut DynPixel, ptrdiff_t, *mut DynCoef, libc::c_int, libc::c_int) -> (),
+>;
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct Dav1dInvTxfmDSPContext {
+    pub itxfm_add: [[itxfm_fn; 17]; 19],
+}
+
+#[cfg(feature = "asm")]
+macro_rules! decl_itx_fn {
+    ($name:ident) => {
+        // TODO(legare): Temporarily pub until init fns are deduplicated.
+        pub(crate) fn $name(
+            dst: *mut DynPixel,
+            dst_stride: ptrdiff_t,
+            coeff: *mut DynCoef,
+            eob: libc::c_int,
+            bitdepth_max: libc::c_int,
+        );
+    };
+
+    ($prefix:ident, $opt:ident) => {
+        paste::paste! {
```

Comment:
Is there a reason why we don't do `use paste::paste;` and just `paste!` here?

Reply:
Mostly for simplicity, since we'd also have to have a `#[cfg(feature = "asm")]` on the import to avoid warnings, so it seemed cleaner to just use the absolute path directly.

Reply:
Ah, that's fine then.  I also noticed in #411 that using non-absolute paths in macros isn't good since it'll resolve where the macro is expanded, so any `use`s have to be put inside the macro.